server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts

```diff
@@ -84,23 +84,18 @@ export default class ServiceProviderReferralsController {
   }
 
   async showReferral(req: Request, res: Response): Promise<void> {
-    const sentReferral = await this.interventionsService.getSentReferral(
-      res.locals.user.token.accessToken,
-      req.params.id
-    )
+    const { accessToken } = res.locals.user.token
+    const sentReferral = await this.interventionsService.getSentReferral(accessToken, req.params.id)
 
     const [intervention, sentBy, expandedServiceUser, conviction, riskInformation] = await Promise.all([
-      this.interventionsService.getIntervention(
-        res.locals.user.token.accessToken,
-        sentReferral.referral.interventionId
-      ),
+      this.interventionsService.getIntervention(accessToken, sentReferral.referral.interventionId),
       this.communityApiService.getUserByUsername(sentReferral.sentBy.username),
       this.communityApiService.getExpandedServiceUserByCRN(sentReferral.referral.serviceUser.crn),
       this.communityApiService.getConvictionById(
         sentReferral.referral.serviceUser.crn,
         sentReferral.referral.relevantSentenceId
       ),
-      this.assessRisksAndNeedsService.getSupplementaryRiskInformation(sentReferral.supplementaryRiskId),
+      this.assessRisksAndNeedsService.getSupplementaryRiskInformation(sentReferral.supplementaryRiskId, accessToken),
     ])
 
     const assignee =
```

Comment:
looks good, but those 2 controllers seems to have identical code, what's different? I think the shared code could be extracted and reused to avoid duplication and having to test in both places.

Reply:
agreed! we have quite a few instances of this... needs some tidy up. if only we had the luxury of time !

Reply:
i feel you!